tracing-mock/src/span.rs

```diff
@@ -760,3 +886,72 @@ impl fmt::Debug for NewSpan {
         s.finish()
     }
 }
+
+impl PartialEq for ExpectedId {
+    fn eq(&self, other: &Self) -> bool {
+        let own_id = self.inner.load(Ordering::Relaxed);
+        let other_id = other.inner.load(Ordering::Relaxed);
+
+        own_id != Self::UNSET && own_id == other_id
+    }
+}
+
+impl Eq for ExpectedId {}
```

Comment:
Nit, this should probably be skipped. Because of `ExpectedId::UNSET` handling in `eq`, this relation is not reflexive so this shouldn't implement `Eq`.

I don't think it's really needed though?

Reply:
You're right about this. The implementation for `PartialEq` isn't reflexive. I added it without thinking because `ExpectedSpan` implements `Eq`. In fact, most of the public structs in `tracing-mock` implement `Eq`.

I don't really want to change this if it's not necessary, so I've change the implementation of `PartialEq` so that it also supports `Eq`.

We were also not actually using the special UNSET != UNSET behavior, and it's probably best not to make that implementation surprising when it doesn't need to be.